lib/Dialect/HW/HWTypes.cpp

```diff
@@ -413,6 +435,17 @@ void UnionType::print(AsmPrinter &odsPrinter) const {
   odsPrinter << ">";
 }
 
+LogicalResult UnionType::verify(function_ref<InFlightDiagnostic()> emitError,
+                                ArrayRef<UnionType::FieldInfo> elements) {
+  llvm::SmallDenseSet<StringAttr> fieldNameSet;
+  fieldNameSet.reserve(elements.size());
+  for (const auto &elt : elements)
+    if (!fieldNameSet.insert(elt.name).second)
+      return emitError() << "duplicate field name '" << elt.name.getValue()
+                         << "' in hw.union type";
+  return success();
```

Comment:
What about printing all errors and then erroring if there was any? This would provide an improved debugging experience.

Reply:
Good point, I'll change it.